src/compiler/xslt/compile/compile-scenario.xsl

```diff
@@ -329,4 +331,16 @@
       <xsl:call-template name="x:compile-child-scenarios-or-expects" />
    </xsl:template>
 
+   <xsl:template name="x:set-up-context" as="element()+">
```

Comment:
- This template is not referenced from the other files. Can you consider prefixing it with `local:` like [this](https://github.com/xspec/xspec/blob/3d1b2e1cd6f93fb14c695ff965a5a875e6503500/src/compiler/xquery/node-constructor/node-constructor.xsl#L119)? (The rule isn't officially documented, but I've refactored the compiler code in that way.)
- The `@as` can be more specific.

```suggestion
   <xsl:template name="local:set-up-context" as="element(xsl:variable)+">
```

Reply:
Addressed in 7af7ae68ddcbdfe64043c56cdd4c6fbb87a91918